internal/resolution/variable_sources/global_constraints/global_constraints_test.go

```diff
@@ -0,0 +1,313 @@
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"bar.io","kind":"Bar","version":"v1"}]`,
+	}),
+	"bundle-11": input.NewEntity("bundle-11", map[string]string{
+		property.TypePackage: `{"packageName": "bar-package", "version": "2.0.0"}`,
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"bar.io","kind":"Bar","version":"v1"}]`,
+	}),
+
+	// test-package-2 required package - no dependencies
+	"bundle-15": input.NewEntity("bundle-15", map[string]string{
+		property.TypePackage: `{"packageName": "test-package-2", "version": "1.5.0"}`,
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"buz.io","kind":"Buz","version":"v1"}]`,
+	}),
+	"bundle-16": input.NewEntity("bundle-16", map[string]string{
+		property.TypePackage: `{"packageName": "test-package-2", "version": "2.0.1"}`,
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"buz.io","kind":"Buz","version":"v1"}]`,
+	}),
+	"bundle-17": input.NewEntity("bundle-17", map[string]string{
+		property.TypePackage: `{"packageName": "test-package-2", "version": "3.16.0"}`,
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"buz.io","kind":"Buz","version":"v1"}]`,
+	}),
+
+	// completely unrelated
+	"bundle-12": input.NewEntity("bundle-12", map[string]string{
+		property.TypePackage: `{"packageName": "unrelated-package", "version": "2.0.0"}`,
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"buz.io","kind":"Buz","version":"v1alpha1"}]`,
+	}),
+	"bundle-13": input.NewEntity("bundle-13", map[string]string{
+		property.TypePackage: `{"packageName": "unrelated-package-2", "version": "2.0.0"}`,
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"buz.io","kind":"Buz","version":"v1alpha1"}]`,
+	}),
+	"bundle-14": input.NewEntity("bundle-14", map[string]string{
+		property.TypePackage: `{"packageName": "unrelated-package-2", "version": "3.0.0"}`,
+		property.TypeChannel: `{"channelName":"stable","priority":0}`,
+		property.TypeGVK:     `[{"group":"buz.io","kind":"Buz","version":"v1alpha1"}]`,
+	}),
+}
+
+var _ = Describe("GlobalConstraintsVariableSource", func() {
+	var (
+		inputVariableSource            *MockInputVariableSource
+		globalConstraintVariableSource *global_constraints.GlobalConstraintsVariableSource
+		ctx                            context.Context
+		entitySource                   input.EntitySource
+	)
+
+	BeforeEach(func() {
+		inputVariableSource = &MockInputVariableSource{}
+		globalConstraintVariableSource = global_constraints.NewGlobalConstraintsVariableSource(inputVariableSource)
+		ctx = context.Background()
+
+		// the entity is not used in this variable source
+		entitySource = &PanicEntitySource{}
```

Comment:
For sake of clarification, the Did we use the `NewGlobalConstraintsVariableSource` shouldn't reach out to the entity source because it's used to enforce GVK and Package uniqueness, which is effectively hardcoded in OLM, right?

Reply:
great call

Reply:
yup - that's right - but it still needs to pass it to its inputVariableSource (and that one _could_ use it)

Reply:
Right, makes sense.